### PR TITLE
Serialization performance.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs [hunk 1]
```diff
@@ -982,7 +982,7 @@ private Type ReadType(EncodingKind kind)
                     var typeName = this.ReadStringValue();
 
                     Type type;
-                    if (!_binder.TryGetType(new TypeKey(assemblyName, typeName), out type))
+                    if (_binder == null || !_binder.TryGetType(new TypeKey(assemblyName, typeName), out type))
```
Comment (tmat) [inline src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs]: Perhaps rename ```_binder``` to ```_binderOpt```?

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @mattwar .  This reduces a layer of indirection in serialization/deserialization .

**Comment by jcouv** [comment]
Is this still proposed for RTM?

**Comment by Pilchie** [comment]
@jcouv - yes, this is a fix to an RPS flagged regression.

File: src/Compilers/Core/Portable/Serialization/ObjectBinder.cs [hunk 2]
```diff
@@ -1,46 +1,93 @@
...
+        private static readonly Action<ObjectWriter, object> s_writer
+            = (w, i) => ((IObjectWritable)i).WriteTo(w);
 
         /// <summary>
         /// Gets a function that writes an object's members to a <see cref="ObjectWriter"/>.
         /// Returns false if the type cannot be serialized.
         /// </summary>
-        public virtual bool TryGetWriter(object instance, out Action<ObjectWriter, object> writer)
-        {          
+        public bool TryGetWriter(Object instance, out Action<ObjectWriter, object> writer)
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Serialization/ObjectBinder.cs]: Nit: Not sure why this is "Object" instead of "object"

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review.

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review (RTW perf improvement).

@CyrusNajmabadi It will be good if you include some measure of the improvement, as I'm sure shiproom will ask once this goes to ask mode.

**Comment by CyrusNajmabadi** [comment]
Unfortunately, the  RPS results indicate that this did not improve things.  @mattwar Do you have any ideas on what we can do here to address the RPS regression?

**Comment by Pilchie** [comment]
Is this the final change?  I thought we ended up finding that it got 60% back, and we wanted to check in to `post-dev15`?

**Pull request closed without merging**